.github/workflows/upstream-dev-ci.yaml

```diff
@@ -0,0 +1,98 @@
+        with:
+          access_token: ${{ github.token }}
+      - uses: actions/checkout@v2
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          channels: conda-forge
+          mamba-version: "*"
+          activate-environment: xarray-tests
+          auto-update-conda: false
+          python-version: ${{ matrix.python-version }}
+      - name: Set up conda environment
+        run: |
+          mamba env update -f ci/requirements/upstream-dev.yml
+          python -m pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple \
+                         --no-deps --pre --upgrade numpy scipy pandas 
+          conda list
+      - name: Run Tests
+        run: |
+          python -m pytest --verbose -rf > output-${{ matrix.python-version }}-log
+
+      - name: Upload artifacts
+        if: "failure()&&(github.event_name == 'schedule')&&(github.repository == 'pydata/xarray')" # Check the exit code of previous step
+        uses: actions/upload-artifact@v2
+        with:
+          name: output-${{ matrix.python-version }}-log
+          path: output-${{ matrix.python-version }}-log
+          retention-days: 5
+
+  report:
+    name: report
+    needs: upstream-dev
+    if: "always()&&(github.event_name == 'schedule')&&(github.repository == 'pydata/xarray')"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
+        with:
+          path: /tmp/workspace/logs
+      - name: Move all log files into a single directory
+        run: |
+          rsync -a /tmp/workspace/logs/output-*/ ./logs
+          ls -R ./logs
+
+      - name: Create an issue
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const fs = require('fs');
+            const files = fs.readdirSync("./logs").filter(fn => fn.endsWith('log'));
+            console.log(files);
+            let logs = '';
+            let file = '';
+            let content = '';
+            for (let i=0; i<files.length; i++) {
+                file = files[i];
+                content = fs.readFileSync(`./logs/${file}`, 'utf8');
+                logs = "\n" + `${logs}` + `\n<details>\n<summary>\nPython ${file.split('-')[1]} Test Summary Info\n</summary>\n\n` + "```bash\n" + `${content}` + "\n" + "```\n</details>";
+              };
+            const title = "⚠️ Nightly upstream-dev CI failed ⚠️"
+            const workflow_url = `https://github.com/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`
+            const issue_body = `[Workflow Run URL](${workflow_url})\n${logs}`
+            github.issues.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: issue_body,
+              title: title,
+              labels: ['CI']
+             })
```

Comment:
I thought the same. One fairly manual resolution to this is to merge something xfail the test, and then a PR to un-xfail the test.

Otherwise it would have to search the existing issues — maybe possible though probably difficult?

Reply:
> I might be missing something (especially since I don't know much about github script and action artifacts), but this looks like it will open a new issue on every run while the fix is being worked on. Is there a way to avoid that?



An issue will be created only when the following condition is met

```yaml
if: "always()&&(github.event_name == 'schedule')&&(github.repository == 'pydata/xarray')"
```

i.e., the workflow run has to be triggered via a `cron schedule`, and this trigger should be active on the `pydata/xarray` repository (which allows us to avoid opening issues on other people's folks). Currently, this condition has some limitations though. For instance, a new issue will be created every day as long as the upstream-dev CI is not passing.

> Otherwise it would have to search the existing issues — maybe possible though probably difficult?

I'm investigating whether there's a straightforward way to keep track of the latest issue created via the GitHub action, and adding comments to an existing issue instead of creating a new issue.

Reply:
> For instance, a new issue will be created every day as long as the upstream-dev CI is not passing.

Exactly, that's what I meant. Sorry for not being clear enough.

> I'm investigating whether there's a straightforward way to keep track of the latest issue created via the GitHub action, and adding comments to an existing issue instead of creating a new issue.

That would be cool, but I would like to make sure we don't post a comment every day, even if the failures don't change. Luckily, `pytest` prints the `short test summary info` section, so we should be able to extract the test ids and do a set comparison with the last comment (or issue, should we decide we need to open a new issue if different tests fail).

Actually, you could upload a file containing the extracted test ids as a artifact and use that to check whether the failed tests changed in the next run. Depending on how we compare, we might even be able to avoid opening a new issue for the remaining failures after merging a partial fix.

Reply:
Personally I would skip opening an issue automatically in the initial PR. I would prefer people to monitor the workflow and report issue in a non automatic way.

If we see that this builds get overlooked, then we add the automatic reporting.

Reply:
Do we think people are going to proactively check the overnight runs? Great if so, and if there's a realistic possibility, fine to try it.